libcxx/test/std/experimental/simd/simd.mask.nonmembers/simd_mask_binary.pass.cpp

```diff
@@ -0,0 +1,79 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14
+
+// <experimental/simd>
+//
+// [simd.mask.nonmembers]
+// friend simd_mask operator&&(const simd_mask& lhs, const simd_mask& rhs) noexcept;
+// friend simd_mask operator||(const simd_mask& lhs, const simd_mask& rhs) noexcept;
+// friend simd_mask operator& (const simd_mask& lhs, const simd_mask& rhs) noexcept;
+// friend simd_mask operator| (const simd_mask& lhs, const simd_mask& rhs) noexcept;
+// friend simd_mask operator^ (const simd_mask& lhs, const simd_mask& rhs) noexcept;
+
+#include "../test_utils.h"
+#include <cassert>
+#include <experimental/simd>
+
+namespace ex = std::experimental::parallelism_v2;
+
+struct CheckSimdMaskBinary {
+  template <class _Tp, class SimdAbi>
+  void operator()() {
+    ex::simd<_Tp, SimdAbi> l_simd(static_cast<_Tp>(5));
+    ex::simd<_Tp, SimdAbi> r_simd([](_Tp i) { return i; });
+
+    const ex::simd_mask<_Tp, SimdAbi> lhs(l_simd == r_simd);
+    const ex::simd_mask<_Tp, SimdAbi> rhs(l_simd > r_simd);
+
+    {
+      ex::simd_mask<_Tp, SimdAbi> logical_and_mask(lhs && rhs);
+      for (size_t i = 0; i < logical_and_mask.size(); ++i) {
+        assert(logical_and_mask[i] == (lhs[i] && rhs[i]));
+      }
```

Comment:
Use `assert_simd_mask_value_correct` to check. Please check all tests for such problems. I will not mark them again